src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -37,24 +37,34 @@ py::dtype get_dtype(const ov::element::Type& ov_type) {
     return ov_type_to_dtype().at(ov_type);
 }
 
-const std::map<std::string, ov::element::Type>& dtype_to_ov_type() {
-    static const std::map<std::string, ov::element::Type> dtype_to_ov_type_mapping = {
-        {"float16", ov::element::f16},
-        {"float32", ov::element::f32},
-        {"float64", ov::element::f64},
-        {"int8", ov::element::i8},
-        {"int16", ov::element::i16},
-        {"int32", ov::element::i32},
-        {"int64", ov::element::i64},
-        {"uint8", ov::element::u8},
-        {"uint16", ov::element::u16},
-        {"uint32", ov::element::u32},
-        {"uint64", ov::element::u64},
-        {"bool", ov::element::boolean},
-        {"bytes_", ov::element::string},
-        {"str_", ov::element::string},
-        {"bytes", ov::element::string},
-        {"str", ov::element::string},
+const std::map<int, ov::element::Type>& dtype_num_to_ov_type() {
+    static const std::map<int, ov::element::Type> dtype_to_ov_type_mapping = {
+        {23, ov::element::f16},  // float16
+        {11, ov::element::f32},  // float32
+        {12, ov::element::f64},  // float64
+        {1, ov::element::i8},    // int8
+        {3, ov::element::i16},   // int16
+#ifdef _WIN32
+        {7, ov::element::i32},  // int32
+        {9, ov::element::i64},  // int64
+#else
+        {5, ov::element::i32},  // int32
+        {7, ov::element::i64},  // int64
```

Comment:
why are we sure that these numbers are valid for different platforms and NumPy versions?